spine-unity/Assets/spine-unity/SkeletonRenderer.cs

```diff
@@ -254,80 +299,163 @@ public class SkeletonRenderer : MonoBehaviour {
-				uvs[vertexIndex + 1] = new Vector2(regionUVs[RegionAttachment.X4], regionUVs[RegionAttachment.Y4]);
-				uvs[vertexIndex + 2] = new Vector2(regionUVs[RegionAttachment.X2], regionUVs[RegionAttachment.Y2]);
-				uvs[vertexIndex + 3] = new Vector2(regionUVs[RegionAttachment.X3], regionUVs[RegionAttachment.Y3]);
+				uvs[vertexIndex].x = regionUVs[RegionAttachment.X1];
+				uvs[vertexIndex].y = regionUVs[RegionAttachment.Y1];
+				uvs[vertexIndex + 1].x = regionUVs[RegionAttachment.X4];
+				uvs[vertexIndex + 1].y = regionUVs[RegionAttachment.Y4];
+				uvs[vertexIndex + 2].x = regionUVs[RegionAttachment.X2];
+				uvs[vertexIndex + 2].y = regionUVs[RegionAttachment.Y2];
+				uvs[vertexIndex + 3].x = regionUVs[RegionAttachment.X3];
+				uvs[vertexIndex + 3].y = regionUVs[RegionAttachment.Y3];
+
+				// Calculate min/max X
+				if (tempVertices[RegionAttachment.X1] < meshBoundsMin.x)
+					meshBoundsMin.x = tempVertices[RegionAttachment.X1];
+				else if (tempVertices[RegionAttachment.X1] > meshBoundsMax.x)
+					meshBoundsMax.x = tempVertices[RegionAttachment.X1];
+				if (tempVertices[RegionAttachment.X2] < meshBoundsMin.x)
+					meshBoundsMin.x = tempVertices[RegionAttachment.X2];
+				else if (tempVertices[RegionAttachment.X2] > meshBoundsMax.x)
+					meshBoundsMax.x = tempVertices[RegionAttachment.X2];
+				if (tempVertices[RegionAttachment.X3] < meshBoundsMin.x)
+					meshBoundsMin.x = tempVertices[RegionAttachment.X3];
+				else if (tempVertices[RegionAttachment.X3] > meshBoundsMax.x)
+					meshBoundsMax.x = tempVertices[RegionAttachment.X3];
+				if (tempVertices[RegionAttachment.X4] < meshBoundsMin.x)
+					meshBoundsMin.x = tempVertices[RegionAttachment.X4];
+				else if (tempVertices[RegionAttachment.X4] > meshBoundsMax.x)
+					meshBoundsMax.x = tempVertices[RegionAttachment.X4];
+
+				// Calculate min/max Y
+				if (tempVertices[RegionAttachment.Y1] < meshBoundsMin.y)
+					meshBoundsMin.y = tempVertices[RegionAttachment.Y1];
+				else if (tempVertices[RegionAttachment.Y1] > meshBoundsMax.y)
+					meshBoundsMax.y = tempVertices[RegionAttachment.Y1];
+				if (tempVertices[RegionAttachment.Y2] < meshBoundsMin.y)
+					meshBoundsMin.y = tempVertices[RegionAttachment.Y2];
+				else if (tempVertices[RegionAttachment.Y2] > meshBoundsMax.y)
+					meshBoundsMax.y = tempVertices[RegionAttachment.Y2];
+				if (tempVertices[RegionAttachment.Y3] < meshBoundsMin.y)
+					meshBoundsMin.y = tempVertices[RegionAttachment.Y3];
+				else if (tempVertices[RegionAttachment.Y3] > meshBoundsMax.y)
+					meshBoundsMax.y = tempVertices[RegionAttachment.Y3];
+				if (tempVertices[RegionAttachment.Y4] < meshBoundsMin.y)
+					meshBoundsMin.y = tempVertices[RegionAttachment.Y4];
+				else if (tempVertices[RegionAttachment.Y4] > meshBoundsMax.y)
+					meshBoundsMax.y = tempVertices[RegionAttachment.Y4];
+
+				// Calculate min/max Z
+				if (z < meshBoundsMin.z)
+					meshBoundsMin.z = z;
+				else if (z > meshBoundsMax.z)
+					meshBoundsMax.z = z;
```

Comment:
We could move `meshBoundsMin/Max.z` outside the loop and use the number of visible attachments to compute it. `z` is `i * zSpacing` but it could change to `i * visibleAttachmentCountSoFar`.